### PR TITLE
Fix cast() function on null values

### PR DESCRIPTION
**Pull request by henridf**

### CODE REVIEW

File: expr/ztests/shape-null-cast.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+zql: |
+   put . = shape(., "{a:string,b:string}")
+
+
```
Comment (nwt) [inline expr/ztests/shape-null-cast.yaml:4]:
```suggestion

```

File: expr/shaper.go [hunk 2]
```diff
@@ -87,6 +87,10 @@ func createOp(in, out *zng.TypeRecord) (op, error) {
 
 func (s op) castPrimitive(in zcode.Bytes, b *zcode.Builder) {
 	pc := LookupPrimitiveCaster(s.castTypes.to)
+	if in == nil {
+		b.AppendNull()
+		return
+	}
 	v, err := pc(zng.Value{s.castTypes.from, in})
```
Comment (nwt) [inline expr/shaper.go:94]: NIt: Keep `pc`'s definition close to its use.
```suggestion
	if in == nil {
		b.AppendNull()
		return
	}
	pc := LookupPrimitiveCaster(s.castTypes.to)
	v, err := pc(zng.Value{s.castTypes.from, in})
```

**Review by nwt — APPROVED** [review]

File: expr/ztests/shape-null-cast.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+zql: |
+   put . = shape(., "{a:string,b:string}")
+input: |
```
Comment (nwt) [inline expr/ztests/shape-null-cast.yaml:3]:
```suggestion
   put . = shape(., "{a:string,b:string}")

input: |
```

**Review by mccanne — APPROVED** [review]
oops

**Pull request merged**